engine/crates/runtime-local/src/hooks.rs

```diff
@@ -15,22 +15,32 @@ use runtime::{
     hooks::{AuthorizedHooks, HeaderMap, Hooks, SubgraphHooks},
 };
 use tracing::instrument;
+pub use wasi_component_loader::{
+    create_log_channel, AccessLogMessage, ChannelLogReceiver, ChannelLogSender, ComponentLoader,
+    Config as HooksWasiConfig,
+};
 use wasi_component_loader::{
-    AuthorizationComponentInstance, GatewayComponentInstance, GuestError, SubgraphComponentInstance,
+    AuthorizationComponentInstance, GatewayComponentInstance, GuestError, SharedContext, SubgraphComponentInstance,
 };
-pub use wasi_component_loader::{ComponentLoader, Config as HooksWasiConfig};
 
 pub struct HooksWasi(Option<HooksWasiInner>);
+
 type Context = Arc<HashMap<String, String>>;
 
 struct HooksWasiInner {
     gateway: Pool<GatewayComponentInstance>,
     authorization: Pool<AuthorizationComponentInstance>,
     subgraph: Pool<SubgraphComponentInstance>,
     hook_latencies: Histogram<u64>,
+    sender: ChannelLogSender,
+    lossy_log: bool,
 }
 
 impl HooksWasiInner {
+    pub fn shared_context(&self, kv: &Arc<HashMap<String, String>>) -> SharedContext {
```

Comment:
I would have just made:
```
#[derive(Clone)]
struct Context {
  kv: Arc<HashMap<String, String>>>,
  sender: ChannelLogSender
}
```
You can add the sender after calling the wasm `on_gateway_request` function, no need for `SharedContext` vs `Context` imo

Reply:
well saying that but I will with authorized at some point... so well, but it's in the future

Reply:
Yeah but this will annoyingly change the interface to the outside a bit too much to my liking. I like that the implementation adds the channel if needed, but the outside doesn't really get it back. I tried that first and I seriously didn't like how it looked.

Reply:
You just have to believe me for now this approach is better. We can reconsider when the feature is working and the big picture is more clear!